php-frontend/src/test/java/org/sonar/php/cfg/ControlFlowGraphTest.java

```diff
@@ -1129,6 +1158,54 @@ public void switch_statement_with_loop_and_continue() {
     );
   }
 
+  @Test
+  public void test_buildCFG() {
+    CompilationUnitTree tree = parse("<?php\n" +
+        "function foo() {\n" +
+        "$expr = function() {echo 'Hello';};\n" +
+        "}" +
+        "echo 'Hello';",
+      PHPLexicalGrammar.COMPILATION_UNIT);
+    FunctionDeclarationTree func = (FunctionDeclarationTree) tree.script().statements().get(0);
+    ExpressionStatementTree expr = (ExpressionStatementTree) func.body().statements().get(0);
+    ControlFlowGraph cfg = ControlFlowGraph.build(func, checkContext);
+    assertThat(cfg.start().elements().get(0)).isEqualTo(expr);
+
+    FunctionExpressionTree funcExpr = ((FunctionExpressionTree) ((AssignmentExpressionTree) expr.expression()).value());
+    StatementTree echo = funcExpr.body().statements().get(0);
+    cfg = ControlFlowGraph.build(funcExpr, checkContext);
+    assertThat(cfg.start().elements().get(0)).isEqualTo(echo);
+
+    StatementTree scriptEcho = tree.script().statements().get(1);
+    cfg = ControlFlowGraph.build(tree.script(), checkContext);
+    assertThat(cfg.start().elements().get(0)).isEqualTo(func);
+    assertThat(cfg.start().elements().get(1)).isEqualTo(scriptEcho);
+  }
+
+  @Test
+  public void test_buildCFG_with_method() {
+    CompilationUnitTree tree = parse("<?php\n" +
+        "class A {\n" +
+        "function foo() {\n" +
+        "echo 'Hello';" +
+        "}" +
+        "abstract function bar();" +
+        "}",
+      PHPLexicalGrammar.COMPILATION_UNIT);
+    ClassDeclarationTree cls = (ClassDeclarationTree) tree.script().statements().get(0);
+    MethodDeclarationTree method = (MethodDeclarationTree) cls.members().get(0);
+    StatementTree echo = ((BlockTree) method.body()).statements().get(0);
+
+    ControlFlowGraph cfg = ControlFlowGraph.build(method, checkContext);
+    assertThat(cfg.start().elements().get(0)).isEqualTo(echo);
+
+    MethodDeclarationTree abstractMethod = (MethodDeclarationTree) cls.members().get(1);
+    cfg = ControlFlowGraph.build(abstractMethod.body(), checkContext);
```

Comment:
that's a weird test, you should pass method, not its body and we will just return `null` without warning.

Reply:
indeed, this was a mistake